src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js

```diff
@@ -0,0 +1,154 @@
+ */
+
+import React, { useMemo } from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { t } from 'ttag';
+import { get } from 'lodash';
+import { COLORS } from '../../../styles/themes';
+import { commonStyles } from '../theme';
+import { onExceptionCaptured } from '../../../actions';
+import { NANOCONTRACT_BLUEPRINTINFO_STATUS as STATUS } from '../../../constants';
+import { FeedbackContent } from '../../FeedbackContent';
+import Spinner from '../../Spinner';
+
+/**
+ * Get method info from registered blueprint data.
+ *
+ * @param {{
+ *   data: Object;
+ * }} blueprint The blueprint info object
+ * @param {string} method The method name to get info from blueprint public methods
+ *
+ * @returns {Object}
+ */
+function getMethodInfoFromBlueprint(blueprint, method) {
+  return get(blueprint.data, `public_methods.${method}`, null);
+}
+
+/**
+ * Get the fallback entries for the method arguments.
+ *
+ * @param {string[]} args A list of argument value
+ *
+ * @returns {[argName: string, value: string][]}
+ *
+ * @example
+ * getFallbackArgEntries([...argValues])
+ * >>> [['Position 0', 'abc'], ['Position 1', '00'], ['Position 2', 123]]
+ */
+function getFallbackArgEntries(args) {
+  return args.map((arg, idx) => [t`Position ${idx}`, arg]);
+}
+
+/**
+ * It renders a list of method arguments for when the Nano Contract executes.
+ *
+ * @param {Object} props
+ * @param {string} props.blueprintId ID of blueprint.
+ * @param {string} props.method Method's name.
+ * @param {string[]} props.ncArgs A list of method's argument.
+ */
+export const NanoContractMethodArgs = ({ blueprintId, method, ncArgs }) => {
+  if (!ncArgs.length) {
+    return null;
+  }
+  const dispatch = useDispatch();
+
+  const blueprintInfo = useSelector((state) => state.nanoContract.blueprint[blueprintId]);
+  // It results a in a list of entries like:
+  // >>> [['oracle_script', 'abc'], ['token_uid', '00'], ['date_last_bet', 123]]
+  // or a fallback like:
+  // >>> [['Position 0', 'abc'], ['Position 1', '00'], ['Position 2', 123]]
+  const argEntries = useMemo(() => {
+    if (blueprintInfo == null || blueprintInfo.status === STATUS.LOADING) {
+      return [];
+    }
+
+    const methodInfo = getMethodInfoFromBlueprint(blueprintInfo, method);
+    if (methodInfo) {
+      return ncArgs.map((arg, idx) => [methodInfo.args[idx].name, arg]);
+    }
+
+    // Send this condition to sentry because it should never happen.
+    // Check any change in the lib or in the fullnode that could cause an impact here.
+    const errMsg = 'Error while getting the argument names of public_methods on blueprint';
+    dispatch(onExceptionCaptured(new Error(errMsg), false));
+
+    // Still render a fallback
+    return getFallbackArgEntries(ncArgs);
+  }, [method, ncArgs, blueprintInfo]);
+
+  // Empty while downloading the bleuprint details
+  const isEmpty = argEntries.length === 0;
```

Comment:
We might have a method without arguments, so you can't show the feedback modal always when the argEntries is empty. You should check the loading status.

Reply:
Oh! At the beginning of the method we have this:
```js
export const NanoContractMethodArgs = ({ blueprintId, method, ncArgs }) => {
  if (!ncArgs.length) {
    return null;
  }
```

It means we don't render the component if the transaction doesn't have arguments to call.